tests/system/src/test/java/test/com/sun/javafx/text/TextLayoutTest.java

```diff
@@ -0,0 +1,521 @@
+/*
+ * Copyright (c) 2012, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.com.sun.javafx.text;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import com.sun.javafx.font.CharToGlyphMapper;
+import com.sun.javafx.font.PGFont;
+import com.sun.javafx.geom.Point2D;
+import com.sun.javafx.geom.RectBounds;
+import com.sun.javafx.scene.text.FontHelper;
+import com.sun.javafx.scene.text.GlyphList;
+import com.sun.javafx.scene.text.TextLine;
+import com.sun.javafx.scene.text.TextSpan;
+import com.sun.javafx.text.PrismTextLayout;
+
+import javafx.scene.text.Font;
+
+public class TextLayoutTest {
+    private static final String J = "\u3041";  // Japanese not complex
+    private static final String D = "\u0907";  // Devanagari complex
+    private static final String T = "\u0E34";  // Thai complex
+
+    private final PrismTextLayout layout = new PrismTextLayout();
+    private final PGFont font = (PGFont) FontHelper.getNativeFont(Font.font("Monaco", 12));
+    private final PGFont font2 = (PGFont) FontHelper.getNativeFont(Font.font("Tahoma", 12));
```

Comment:
They may not be present on all platforms, but that doesn't matter. All that matters is that they're present in our 3 build environments (linux, mac, windows), and since the build is succeeding, they must be present :)

If they're not present, the test will simply fail.

Reply:
However... it does mean these fonts need to be present on your local environment as well if you want to build with test succeeding.  I don't know if we can make that a requirement.  I could add JUnit 5 `assume` here, but then it may magically stop executing the test if these fonts were ever removed from our build environment.